openhands/core/agent/agent/agent.py

```diff
@@ -81,6 +87,24 @@ def step(
         state: ConversationState,
         on_event: ConversationCallbackType,
     ) -> None:
+        # Check if we have pending actions to execute (implicit confirmation)
+        # Only execute pending actions if we haven't created any actions in this run
+        if (
+            self.confirmation_mode
+            and self._pending_actions
```

Comment:
@openhands do you need to track `_pending_actions`? 

Could you find the latest action in the history and check whether it has a matching observation instead? That way we avoid having to track extra states